misc/python/materialize/mzcompose.py

```diff
@@ -64,6 +64,36 @@
 )
 
 
+class LintError:
+    def __init__(self, file: Path, message: str):
+        self.file = file
+        self.message = message
+
+    def __str__(self) -> str:
+        return f"{self.file.relative_to(Path.cwd())}: {self.message}"
+
+    def __lt__(self, other: "LintError") -> bool:
+        return (self.file, self.message) < (other.file, other.message)
+
+
+def lint_composition(path: Path, composition: Any, errors: List[LintError]) -> None:
+    for (name, service) in composition["services"].items():
+        if service.get("mzbuild") == "materialized":
+            lint_materialized_service(path, service, errors)
+
+
+def lint_materialized_service(
+    path: Path, service: Any, errors: List[LintError]
+) -> None:
+    if "--disable-telemetry" not in service.get("command", "").split():
```

Comment:
not sure if the `.split()` is necessary, but it's not necessarily hurting.

Reply:
Oh, I forgot to respond to this. The idea is to only accept "foo --disable-telemetry" but not "typo--disable-telemetry".